File: app/src/main/java/com/studyasist/data/local/entity/GoalItem.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "goal_items",
    foreignKeys = [
        ForeignKey(
            entity = Goal::class,
            parentColumns = ["id"],
            childColumns = ["goalId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["goalId"])]
)
data class GoalItem(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val goalId: Long,
    val subject: String,
    val chapterList: String,
    val targetHours: Int? = null
)

File: StudyAsist-iOS/StudyAsist/Models/GoalItem.swift
//
//  GoalItem.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class GoalItem {
    @Attribute(.unique) var id: Int64
    var goalId: Int64
    var subject: String
    var chapterList: String
    var targetHours: Int?
    
    init(
        id: Int64 = 0,
        goalId: Int64,
        subject: String,
        chapterList: String,
        targetHours: Int? = nil
    ) {
        self.id = id
        self.goalId = goalId
        self.subject = subject
        self.chapterList = chapterList
        self.targetHours = targetHours
    }
}
